SICP_book/chapter_1_3.metta

```diff
@@ -0,0 +1,485 @@
+;needed for putting lambda functions as input to functions
+(: Lambda (-> $a $t))
+(: lambda (-> Variable Atom Lambda))
```

Comment:
1. It is better to inline `Lambda` into `lambda` here and below to not introduce separate type without needness: `(: lambda (-> Variable Atom (-> $a $t)))`
2. I would also rename `lambda` to `lambda1` to align naming with other definitions
3. It is better to mention that the only reason why different definitions of `lambdaX` are needed is that we would like to make lambda's interchangeable with other functions. Otherwise we could define universal lambda like this `(= ((lambda $vars $body) $vals) (let $vars $vals $body))` and use it like this `(let $sum (lambda ($a $b $c) (+ (+ $a $b) $c)) ($sum (1 2 3))`